lib/types/src/transaction/encode.rs

```diff
@@ -23,6 +26,21 @@ impl<T: L1TxType> ZksyncOsEncode for L1Envelope<T> {
     }
 }
 
+// todo: to be reused from basic bootloader code
+pub const BOOTLOADER_FORMAL_ADDRESS: B160 = B160::from_limbs([0x8001, 0, 0]);
+
+impl<T: SystemTxType> ZksyncOsEncode for SystemTransactionEnvelope<T> {
+    fn encode(self) -> EncodedTx {
+        let mut rlp_body = Vec::new();
+        Encodable::encode(&self.inner, &mut rlp_body);
+        let mut out = Vec::with_capacity(1 + rlp_body.len());
+        out.push(T::TX_TYPE);
+        out.extend_from_slice(&rlp_body);
+        let from = Address::from_slice(&BOOTLOADER_FORMAL_ADDRESS.to_be_bytes::<20>());
+        EncodedTx::Rlp(out, from)
+    }
+}
```

Comment:
I'd try to implement `Encodable2718` for `SystemTransactionEnvelope` so that this implementation looks like `EncodedTx::Rlp(self.encoded_2718(), BOOTLOADER_FORMAL_ADDRESS)`